modules/openapi-generator/src/main/resources/go-experimental/model.mustache

```diff
@@ -35,14 +35,14 @@ type {{classname}} struct {
 {{#description}}
 	// {{{description}}}
 {{/description}}
-	{{name}} *{{{dataType}}} `json:"{{baseName}},omitempty"{{#withXml}} xml:"{{baseName}}{{#isXmlAttribute}},attr{{/isXmlAttribute}}"{{/withXml}}{{#vendorExtensions.x-go-custom-tag}} {{{.}}}{{/vendorExtensions.x-go-custom-tag}}`
-{{#isNullable}}	isExplicitNull{{name}} bool `json:"-"{{#withXml}} xml:"-"{{/withXml}}`{{/isNullable}}
+	{{name}} {{^required}}*{{/required}}{{{dataType}}} `json:"{{baseName}}{{^required}}{{^isNullable}},omitempty{{/isNullable}}{{/required}}"{{#withXml}} xml:"{{baseName}}{{#isXmlAttribute}},attr{{/isXmlAttribute}}"{{/withXml}}{{#vendorExtensions.x-go-custom-tag}} {{{.}}}{{/vendorExtensions.x-go-custom-tag}}`
```

Comment:
I see two cases where your implementation is problematic:

* How would this work with a field that is required and nullable? AFAICS you'd have
  `Foo int ``json:"foo"`` `, but there's no way to serialize the explicit `null` in this scenario AFAICS.
* For scenario where you have a non-required nullable field, how do you distinguish when the field is just null because it was initialized that way versus when the user explicitly set it null?

Reply:
As for the `required` and `nullable` field, you won't even be able to assign nil, because the type is not pointer - it's `Foo int`, not `Foo *int`.

> but at least we don't introduce private members to what should be data only struct.

Why should it be a data only structure?

I feel like this is a sacrifice of feature completeness for getting a nicer code, which I'm not in favor of.

Reply:
(Just to add to my previous point, I'd love to be able to simplify this and I'm all for someone doing that. I just don't think that this PR is sufficient, as it sacrifices feature completeness for a bit less boilerplate (which is generated anyway) and a bit of improvement for GC which I honestly don't think would be noticable. Maybe there's a way to modify this PR to make it cover all the cases which I don't see, but I'd be more than happy to give it a thumbsup if you found it.)

Reply:
You are correct, it should be pointer type, I will add it.

Regarding copying, if you copy subset of fields instead of whole object information about field being explicitly null is lost. I also think that gc pressure shouldn't be dismissed here, on every marshaling map is allocated and each value is converted into heap-referenced GC object.

Reply:
Coming to think about it - we could simply generate Nullable types and Ptr methods for object schemas and this way support explicit null.

Reply:
So right now you have

```
{{^required}}{{^isNullable}}*{{/isNullable}}{{/required}}{{{dataType}}}
```

but if you want to export all nullable fields as pointers (as your last commit message suggests), you should do

```
{{#isNullable}}*{{/isNullable}}{{{dataType}}}
```

Am I missing somethings?

> 

Coming to think about it - we could simply generate Nullable types and Ptr methods for object schemas and this way support explicit null.

I honestly don't know what you mean by that. Could you provide an example of what this would look like?

Reply:
Added Nullable types to the templates. I need to probably override dataType in codegen for nullable types. I need to add Get/Set operations for nullable types and SetExplicitNull method.

This approach guarantees that if we copy field value the information about explicit null is not lost.